### PR TITLE
LibC: openpty error handling update

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: Userland/Libraries/LibC/pty.cpp [hunk 1]
```diff
@@ -56,12 +56,22 @@ int openpty(int* amaster, int* aslave, char* name, const struct termios* termp,
         return -1;
     }
     if (termp) {
-        // FIXME: error handling
-        tcsetattr(*aslave, TCSAFLUSH, termp);
+        if (tcsetattr(*aslave, TCSAFLUSH, termp) == -1) {
+            int error = errno;
```
Comment (gunnarbeutner) [inline Userland/Libraries/LibC/pty.cpp:60]: That reads an uninitialized variable (`errno`) which just so happens to shadow the global `errno` variable.

File: Userland/Libraries/LibC/pty.cpp [hunk 2]
```diff
@@ -56,12 +56,22 @@ int openpty(int* amaster, int* aslave, char* name, const struct termios* termp,
...
+            close(*amaster);
+            errno = error;
+            return -1;
+        }
     }
     if (winp) {
-        // FIXME: error handling
-        ioctl(*aslave, TIOCGWINSZ, winp);
+        if (ioctl(*aslave, TIOCGWINSZ, winp) == -1) {
+            int error = errno;
```
Comment (gunnarbeutner) [inline Userland/Libraries/LibC/pty.cpp:69]: Likewise.

**Review by gunnarbeutner — CHANGES_REQUESTED** [review]

**Comment by gunnarbeutner** [comment]
Sorry about the (now deleted) comment. I thought both variables were named `errno`. My bad. :)

**Comment by devnexen** [comment]
I also did not drink my coffee too :-)

**Pull request merged**